simplesat/dependency_solver.py

```diff
@@ -216,6 +221,87 @@ def simplify_requirements(packages, requirements):
     return simple_requirements
 
 
+def minimal_unsatisfiable_subset(clauses, callback):
+    """
+    Given a set of clauses, find a minimal unsatisfiable subset (an
+    unsatisfiable core)
+
+    A set is a minimal unsatisfiable subset if no proper subset is
+    unsatisfiable.  A set of clauses may have many minimal unsatisfiable
+    subsets of different sizes.
+
+    Parameters
+    ----------
+    clauses : seq
+        Iterator over a set of clauses. Clause should follow the format
+        expected by callback
+    callback : callable
+        A function of the form f: <clauses subset> -> bool, returning True if
+        the given subset of clauses is satisfiable, False otherwise. It should
+        follow the constraint A <= B iff (f(B) <= f(A)), with False < True.
+
+    Note
+    ----
+    Function adapted from conda (conda.logic as of 4.0.5), itself implemented
+    from http://www.slideshare.net/pvcpvc9/lecture17-31382688. The conda
+    function mentions the following:
+
+    We do a binary search on the clauses by splitting them in halves A and B.
+    If A or B is UNSAT, we use that and repeat. Otherwise, we recursively check
+    A, but each time we do a sat query, we include B, until we have a minimal
+    subset A* of A such that A* U B is UNSAT. Then we find a minimal subset B*
+    of B such that A* U B* is UNSAT. Then A* U B* will be a minimal
+    unsatisfiable subset of the original set of clauses.
+
+    Proof: If some proper subset C of A* U B* is UNSAT, then there is some
+    clause c in A* U B* not in C. If c is in A*, then that means (A* - {c}) U
```

Comment:
I found the use of `c` here confusing--I initially read this as `c \in C`. Could you use a different letter?